packages/components/src/Slider/Slider.component.js

```diff
@@ -0,0 +1,116 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import { translate } from 'react-i18next';
+import RcSlider from 'rc-slider';
+import rcSliderTheme from 'rc-slider/assets/index.css'; // eslint-disable-line no-unused-vars
+
+import { Icon } from '../';
+import I18N_DOMAIN_COMPONENTS from '../constants';
+import { DEFAULT_I18N } from '../translate';
+import theme from './Slider.scss';
+
+function noOp() {}
+
+/**
+ * This function render an icon component
+ * @param {object} icon - Icons props ( selected & name )
+ * @param {number} key - index of the icon for the key
+ */
+function renderIcons(icon, key) {
+	return <Icon name={icon.name} className={icon.selected ? theme.selected : null} key={key} />;
+}
+
+/**
+ * This function give the selected icon position if there is more than 1 icon
+ * @param {array} icons - array of icons
+ * @param {number} value - current value of the slider
+ * @param {number} min - minimum value of the slider
+ * @param {number} max - maximum value of the slider
+ */
+export function getSelectedIconPosition(icons, value, min, max) {
+	if (icons && Array.isArray(icons) && icons.length > 1) {
+		const interval = (max - min) / (icons.length - 1);
+		const div = Math.floor(value / interval);
+		const rest = value % interval;
+		return rest < interval / 2 ? div : div + 1;
+	}
+	return -1;
+}
+
+/**
+ * This function allow to get the icons components
+ * @param {array} icons - array of icons
+ * @param {number} value - current value of the slider
+ * @param {number} min - minimum value of the slider
+ * @param {number} max - maximum value of the slider
+ */
+function getIcons(icons, value, min, max) {
+	const position = getSelectedIconPosition(icons, value, min, max);
+	const iconsSelected = icons.map((icon, index) => ({ name: icon, selected: index === position }));
+	return iconsSelected.map(renderIcons);
+}
```

Comment:
```
function getIcons(icons, value, min, max) {
	const position = getSelectedIconPosition(icons, value, min, max);
	return icons.map((icon, index) => (
		<Icon
			name={icon.name}
			className={index === position ? theme.selected : null}
			key={index}
		/>
	));
}
```
This avoids :
* an object creation per icon to store the selected boolean
* a renderIcon() function that only return an <Icon> without any other logic